server/schedule/plan/plan.go

```diff
@@ -16,4 +16,63 @@ package plan
 
 // Plan is the basic unit for both scheduling and diagnosis.
 // TODO: for each scheduler/checker, we can have an individual definition but need to implement the common interfaces.
-type Plan interface{}
+type Plan interface {
+	GetStep() int
+	GetStatus() *Status
+
+	Clone(ops ...Option) Plan // generate plan for clone option
+	GenerateCoreResource(interface{})
+	SetStatus(*Status)
+}
+
+// Collector is a plan collector
+type Collector struct {
+	basePlan           Plan
+	unschedulablePlans []Plan
+	schedulablePlans   []Plan
+	enable             bool
+}
+
+// NewCollector returns a new Collector
+func NewCollector(enable bool, plan Plan) *Collector {
```

Comment:
Do we need to create it if the dryrun is false?

Reply:
I think we can create. The advantage is that we don't have to check nil in related code.

Reply:
Then, we need to create it every time `Schedule` is called.